src/Fixer/PhpUnit/PhpUnitTestClassRequiresCoversFixer.php

```diff
@@ -66,90 +66,87 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
     {
         $phpUnitTestCaseIndicator = new PhpUnitTestCaseIndicator();
 
-        for ($index = $tokens->count() - 1; $index >= 0; --$index) {
-            if (!$tokens[$index]->isGivenKind(T_CLASS)) {
-                continue;
-            }
-
-            $prevIndex = $tokens->getPrevMeaningfulToken($index);
+        foreach ($phpUnitTestCaseIndicator->findPhpUnitClasses($tokens) as $indexes) {
```

Comment:
this used to loop from the last token to the first, no it is the other way around, are you sure this is without issues when the fixer add a lot of tokens?

Reply:
This fixer should not add much tokens, but it has the same issue as https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3745#discussion_r190040263, so it might be best to wait on the results of that.

Reply:
please update after https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3745/files#diff-60a412e1b3e8bf5c206429800960dd55R58 :)

Reply:
ping @BackEndTea this one is close to merging with a little tweak I think :)

Reply:
The default of `findPhpUnitClasses` is to start at the bottom right now. Should i add true as a second argument to make this more explicit here?

Reply:
ah nice, all good than :+1:  :)